lib/pre-commit/utils/staged_files.rb

```diff
@@ -33,7 +33,10 @@ def staged_files_git_all
     private
       # from https://github.com/djberg96/ptools/blob/master/lib/ptools.rb#L90
       def binary?(file)
-        s = (File.read(file, File.stat(file).blksize) || "").split(//)
+        bytes = File.stat(file).blksize
+        bytes = 4096 if bytes > 4096
+        s = (File.read(file, bytes) || "")
+        s = s.encode('US-ASCII', :undef => :replace).split(//)
```

Comment:
I'm just updating the code snippet used in `ptools` found here https://github.com/djberg96/ptools/blob/master/lib/ptools.rb#L90

I'm certainly open to using the approach that works best for the most people.